src/sagemaker/utils.py

```diff
@@ -284,10 +284,16 @@ def download_file(bucket_name, path, target, sagemaker_session):
 
 
 class DeferredError(object):
-    """Stores an exception and raises it at a later time anytime this
+    """Stores an exception and raises it at a later time if this
     object is accessed in any way.  Useful to allow soft-dependencies on imports,
     so that the ImportError can be raised again later if code actually
-    relies on the missing library.
+    relies on the missing library.  Example usage:
```

Comment:
you could do an example section instead, e.g.

```
Examples:
    try:
        import obscurelib
    except ImportError as e:
        logging.warning("Failed to import obscurelib. Obscure features will not work.")
        obscurelib = DeferredError(e)
```

reference: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html